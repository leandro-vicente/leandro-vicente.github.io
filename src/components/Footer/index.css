footer {
  left: 50%;
  transform: translateX(-50%);
  position: absolute;
  bottom: 0px;
  width: 100%;
  text-align: center;
  overflow: hidden;
}

footer div.border {
  width: 60px;
  position: absolute;
  top: 0px;
  left: 50%;
  transform: translateX(-50%);
  border-top: 1px solid;
}

footer address {
  font-family: 'Open Sans', sans-serif;
  font-style: normal;
  font-weight: 100;
  position: relative;
  margin-bottom: 10px;
  margin-top: 5px;
  text-decoration: none;
  left: 50%;
  transform: translateX(-50%);
  padding: 0px 10px;
  width: 50%;
  
}