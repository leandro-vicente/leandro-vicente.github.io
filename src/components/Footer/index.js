import React from 'react'
import PropTypes from 'prop-types'
import './index.css'

const Footer = (props) => {
  const { address } = props
  return(
    <React.Fragment>
      <footer>
        <div className='border'></div>
        <address>
          {address}
        </address>
      </footer>
    </React.Fragment>
  )
}

Footer.propTypes = {
  address: PropTypes.string.isRequired
}

export default Footer