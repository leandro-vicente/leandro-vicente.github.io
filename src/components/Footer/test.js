import React from 'react'
import Footer from './index.js'
import { shallow } from 'enzyme'

describe('Footer', () => {
  it('renders as expected', () => {
    const footer = shallow(<Footer address='BH, Brasil' />)
    expect(footer).toMatchSnapshot()
  })
})