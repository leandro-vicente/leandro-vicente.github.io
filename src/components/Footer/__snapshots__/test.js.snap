// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    address="BH, Brasil"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <footer>
        <div
          className="border"
        />
        <address>
          BH, Brasil
        </address>
      </footer>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="border"
          />,
          <address>
            BH, Brasil
          </address>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "border",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BH, Brasil",
          },
          "ref": null,
          "rendered": "BH, Brasil",
          "type": "address",
        },
      ],
      "type": "footer",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <footer>
          <div
            className="border"
          />
          <address>
            BH, Brasil
          </address>
        </footer>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="border"
            />,
            <address>
              BH, Brasil
            </address>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "border",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BH, Brasil",
            },
            "ref": null,
            "rendered": "BH, Brasil",
            "type": "address",
          },
        ],
        "type": "footer",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
